hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -370,10 +371,20 @@ private Builder generateMigrationStateChangesSidecar(
         IterableContractValue iterableValue;
         while (maxNumberOfKvPairsToIterate > 0 && contractStorageKey != null) {
             iterableValue = contractStorageMap.get(contractStorageKey);
+            final var keyAsBigInteger = contractStorageKey.getKeyAsBigInteger();
```

Comment:
Performance wise this has the potential to be a big bottleneck There can be multiple tuweni bytes objects created as well as biginteger objects created, each with their own backing array.

Contract storage key should go to a byte array once via `toByteArray()` (not the unsafe variant), figure out how many leading zeros there are in that native array, then use `com.google.protobuf.UnsafeByteOperations#unsafeWrap(byte[], int, int)` to wrap the byte string.  This will cut the object churn in about half.

However since this is migration code and will be xecuted once, and we are not seeing performance issues in the tests, this is a truly optional fix.

Reply:
Changed this now, just to be safe --- I think we are still concerned about the `handleTransaction()` time of the transaction that will trigger the migration